packages/flutter/lib/src/gestures/long_press.dart

```diff
@@ -197,25 +236,51 @@ class LongPressGestureRecognizer extends PrimaryPointerGestureRecognizer {
           invokeCallback<void>('onLongPressEnd', () {
             onLongPressEnd(LongPressEndDetails(
               globalPosition: event.position,
+              buttons: _initialButtons,
             ));
           });
         }
-        _longPressAccepted = false;
       } else {
+        // Pointer is lifted before timeout.
         resolve(GestureDisposition.rejected);
       }
-    } else if (event is PointerDownEvent || event is PointerCancelEvent) {
+      _reset();
+    } else if (event is PointerCancelEvent) {
+      _reset();
+    } else if (event is PointerDownEvent) {
       // The first touch.
-      _longPressAccepted = false;
       _longPressOrigin = event.position;
-    } else if (event is PointerMoveEvent && _longPressAccepted && onLongPressMoveUpdate != null) {
-      invokeCallback<void>('onLongPressMoveUpdate', () {
-        onLongPressMoveUpdate(LongPressMoveUpdateDetails(
-          globalPosition: event.position,
-          offsetFromOrigin: event.position - _longPressOrigin,
-        ));
-      });
+      _initialButtons = event.buttons;
+    } else if (event is PointerMoveEvent) {
+      if (event.buttons != _initialButtons) {
+        resolve(GestureDisposition.rejected);
+        stopTrackingPointer(primaryPointer);
+      } else if (_longPressAccepted && onLongPressMoveUpdate != null) {
+        invokeCallback<void>('onLongPressMoveUpdate', () {
+          onLongPressMoveUpdate(LongPressMoveUpdateDetails(
+            globalPosition: event.position,
+            offsetFromOrigin: event.position - _longPressOrigin,
+            buttons: _initialButtons,
+          ));
+        });
+      }
+    }
+  }
+
+  void _reset() {
+    _longPressAccepted = false;
+    _longPressOrigin = null;
+    _initialButtons = null;
+  }
+
+  @override
+  void resolve(GestureDisposition disposition) {
+    if (_longPressAccepted && disposition == GestureDisposition.rejected) {
+      // This can happen if the gesture has been terminated. For example when
```

Comment:
I'm getting confused: Is it "terminated", "canceled", or "rejected"?  Perhaps take a look to see that you're using consistent terminology, or if they really mean different things, then clarify. Or maybe I'm just confused. :-)

Reply:
Yes, I might have used too many terms than needed, and on a second thought, "cancel" alone might be enough. I'll check all occurrences and try to limit them to cancel.

Reply:
Fixed.